Add ERC: Zero-knowledge proof metadata

zwowo1997 opened the pull request:
ZKMeta Metadata Interface, a minimal and proof-system-agnostic standard for exposing zero-knowledge proof metadata to Ethereum smart contracts and tooling.

zwowo1997 committed:
Add files via upload

zwowo1997 committed:
Add erc-0.md file

zwowo1997 committed:
Merge branch 'master' into master

abcoathup commented on ERCS/erc-0.md:
```diff
@@ -0,0 +1,178 @@
+---
+eip: 0
```
```suggestion
eip: 8084
```
Assigning next sequential EIP/ERC/RIP number.
Numbers are assigned by editors & associates.

Please also update the filename.

zwowo1997 committed:
Update ERCS/erc-0.md

Co-authored-by: Andrew B Coathup <28278242+abcoathup@users.noreply.github.com>

zwowo1997 committed:
change title

Co-authored-by: Andrew B Coathup <28278242+abcoathup@users.noreply.github.com>

zwowo1997 committed:
Add new ERCS document erc-8084.md

SamWilsn commented on ERCS/erc-8084.md:
```diff
@@ -0,0 +1,178 @@
+
+## Specification
+
+The keywords “MUST”, “MUST NOT”, “SHOULD”, and “MAY” are as in RFC 2119.
```
```suggestion
The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://www.rfc-editor.org/rfc/rfc2119) and [RFC 8174](https://www.rfc-editor.org/rfc/rfc8174).
```

Just use the wording from the template.

SamWilsn commented on ERCS/erc-8084.md:
```diff
@@ -0,0 +1,178 @@
+
+## Copyright
+
+CC0
```
```suggestion
Copyright and related rights waived via [CC0](../LICENSE.md).
```

Use the exact waiver from the template.

zwowo1997 commented on ERCS/erc-8084.md:
````diff
@@ -0,0 +1,178 @@
+```solidity
+/// @title ERC-ZKMeta Interface
+interface IZKMetadata {
+    /// Emitted when any getter below would return a different value than before.
````
Good catch! changed to: Emitted when the stored circuit metadata is modified.

zwowo1997 commented on ERCS/erc-8084.md:
```diff
@@ -0,0 +1,178 @@
+- Tooling MUST verify that `publicInputsSchemaHash()` matches the fetched schema document.
+- Tooling SHOULD verify that the verification key’s content hash matches `verificationKeyURI()`’s hash fragment/CID.
+- Unknown `proofSystem()` identifiers SHOULD be treated as unsupported.
+- Emit `CircuitMetadataUpdated` atomically with metadata changes to avoid indexer races.
```
Moved rules to Requirement section. Re-wrote security considerations

zwowo1997 committed:
Revise ZKMeta EIP based on SamWilsn comments

Updated the description and abstract sections for clarity and detail. Enhanced the security considerations and recommendations for better understanding.

zwowo1997 reviewed (commented):
Thanks for the comments, Sam. I have revised the draft for your review

zwowo1997 committed:
Refine ZKmeta description and enhance references

Updated the description of ZKmeta for clarity and added a link to ERC-1923 for context.

zwowo1997 committed:
Refine ZKMeta EIP description and motivation

Removed redundant motivation section and improved description clarity.

zwowo1997 committed:
Merge branch 'master' into master
